### PR TITLE
Bug fix for the schedule after the update_optimizer

### PR DESCRIPTION
**Pull request by oguzhanbsolak**
After the update_optimizer, we need to reflect the updated optimizer to the compression scheduler.  Otherwise, the scheduler uses the old optimizer, and lr is not scheduled correctly.  This PR fixes the issue by traversing each policy and replacing the old optimizer object with the new one.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -128,6 +128,7 @@
 
 def main():
     """main"""
+    # pylint: disable=too-many-branches
```
Comment (rotx-maxim) [inline train.py:131]: please remove and fix limits in .pylintrc and .github/linters/.python-lint

**Review by rotx-maxim — CHANGES_REQUESTED** [review]
Please remove line 131 and fix limits in .pylintrc and .github/linters/.python-lint instead

File: train.py (hunk 1 above)
Comment (oguzhanbsolak) [reply train.py]: Realized in the latest commit.

**Review by rotx-maxim — CHANGES_REQUESTED** [review]
Please also change .pylintrc

**Review by rotx-maxim — CHANGES_REQUESTED** [review]
Shouldn't this be:

           if compression_scheduler:
                for ep, _ in enumerate(compression_scheduler.policies):
                    for pol in compression_scheduler.policies[ep]:
                        for attr_key in dir(pol):
                            attr = getattr(pol, attr_key)
                            if hasattr(attr, 'optimizer'):
                                attr.optimizer = optimizer

**Comment by rotx-maxim** [comment]
Has this been tested with anything other than lr_scheduler?

**Comment by oguzhanbsolak** [comment]
Hi Robert,

> Has this been tested with anything other than lr_scheduler?


I tested this change with lr_schedulers and RelationBasedKDPolicy policies. In the current train.py structure, we are only using KnowledgeDistillationPolicy, RelationBasedKDPolicy, and lr_scheduler. From them, only lr_scheduler has the "optimizer" attribute. Therefore, this change shouldn't affect other policies.


> Shouldn't this be:
> 
> ```
>        if compression_scheduler:
>             for ep, _ in enumerate(compression_scheduler.policies):
>                 for pol in compression_scheduler.policies[ep]:
>                     for attr_key in dir(pol):
>                         attr = getattr(pol, attr_key)
>                         if hasattr(attr, 'optimizer'):
>                             attr.optimizer = optimizer
> ```

When we don't define a scheduler policy, the train.py creates a default compression_scheduler. Therefore, there will be always a compression_scheduler at the state that this snippet will be run, and I believe we don't need to check if there is a compression_scheduler. 

For readability your suggested version seems much better, so I am planning to push a new commit as in below:
``` for ep, _ in enumerate(compression_scheduler.policies):
                for pol in compression_scheduler.policies[ep]:
                    for attr_key in dir(pol):
                        attr = getattr(pol, attr_key)
                        if hasattr(attr, 'optimizer'):
                            attr.optimizer = optimizer
```

**Review by rotx-maxim — APPROVED** [review]
